distributed/tests/test_client_executor.py

```diff
@@ -216,21 +216,25 @@ def test_retries(client):
     with client.get_executor(retries=5, pure=False) as e:
         future = e.submit(varying(args))
         assert future.result() == 42
+        del future
```

Comment:
These seem redundant to me?

Reply:
Without this, the future is held on worker, increasing `nbytes` and changing which worker we pick. (This test is _highly_ dependent on how the worker selection logic works because it runs tasks which change their outputs depending on how many times they've been called on a particular worker; if the order of worker selection changes, the number of retries it takes to get a failing/non-faililng result changes.)

Reply:
Ah. You're relying on the release message to deterministically reach the scheduler before the next submit message. Got it.